javascript/javascript1/week4/VoiceAssistant.js

```diff
@@ -0,0 +1,180 @@
+const commandHistory = []
+let curUser = ""
+const todos = []
+
+function getReply(command)
+{
+// save the name and respond with a message
+    //Checking whether the user already indtroduced 
+    let found = false;
+    for(item in commandHistory) 
+    {
+        if(commandHistory[item] === command.toLowerCase()) 
+        {
+            found = true;
+            break;
+        }
+    }
+    //If user already exists print a message else save the user name and set current user
+    if(command.toLowerCase().startsWith("hello my name is"))
+     {
+   
+        curUser = command.split(" ").pop()
+        if(curUser == "")
+        {
+            console.log(`No name found. Enter a valid name`)
+        }
+       else if(found)
+        {
+            console.log(`Hello ${curUser}, we know you already`)
+        } 
+        else 
+        {
+            commandHistory.push(command.toLowerCase()) ;
+            console.log(`nice to meet you ${curUser}`) ;
+        }
+    } 
+// If asking for name, return current user
+    else if(command.toLowerCase().startsWith("what is my name")) 
+    { 
+        if (curUser.length) 
+        {
+            console.log(`Your name is: ${curUser}`);
+        } 
+        else 
+        {
+            console.log(`Sorry we don't recognize you`);
+        }
+    } 
+//Add todo list items
+    else if(command.toLowerCase().startsWith("add") &&
+    command.toLowerCase().endsWith("to my todo")) 
+    {
+
+        const task = command.replace("Add ", "").replace(" to my todo", "");
```

Comment:
This is nice.
I'd use `lowerCaseCommand` to replace here. That way, you don't have to care about lowercase/uppercase letters.